### ISSUE TITLE
bert OutOfRangeError

### ISSUE THREAD

**Issue by zgmming** [liyibo/text-classification-demos#1]
ubuntu16.04+python3.5 +tensorflow-gpu 1.11.0
OutOfRangeError

Iter:    937, Train Loss:  0.011, Train Acc: 100.00%, Val Loss:    0.0, Val Acc:   0.00%, Time: 2019-02-25 19:52:33.882878 
Iter:    938, Train Loss: 0.00036, Train Acc: 100.00%, Val Loss:    0.0, Val Acc:   0.00%, Time: 2019-02-25 19:52:33.945808 
End of sequence
	 [[{{node IteratorGetNext}} = IteratorGetNext[output_shapes=[[?,32], [?,32], [?], [?,32]], output_types=[DT_INT32, DT_INT32, DT_INT32, DT_INT32], _device="/job:localhost/replica:0/task:0/device:CPU:0"](OneShotIterator)]]

Caused by op 'IteratorGetNext', defined at:
  File "/home/kelvin/deeplearning/text-classification-demos/bert_model.py", line 337, in <module>
    main()
  File "/home/kelvin/deeplearning/text-classification-demos/bert_model.py", line 223, in main
    train_batch = train_iterator.get_next()
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/data/ops/iterator_ops.py", line 420, in get_next
    name=name)), self._output_types,
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/gen_dataset_ops.py", line 2069, in iterator_get_next
    output_shapes=output_shapes, name=name)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/util/deprecation.py", line 488, in new_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 3272, in create_op
    op_def=op_def)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 1768, in __init__
    self._traceback = tf_stack.extract_stack()

OutOfRangeError (see above for traceback): End of sequence
	 [[{{node IteratorGetNext}} = IteratorGetNext[output_shapes=[[?,32], [?,32], [?], [?,32]], output_types=[DT_INT32, DT_INT32, DT_INT32, DT_INT32], _device="/job:localhost/replica:0/task:0/device:CPU:0"](OneShotIterator)]]

Process finished with exit code 0

**Comment by liyibo** [comment]
sorry about the bug，this is because the number of train_tf_record is too small, so you can repeat the tf_record by use bigger num in “train_data = read_data(train_record, train_batch_size, True, 3)”，such as 5 or 10, and you can smaller the val steps in " if train_steps % 1000 == 0:",such as 100 or 500, it's depends on the number of your train_data and val data

**Comment by tectal** [comment]
@liyibo 我也遇到同样的问题，按照上面的操作（如下），问题没有得到解决，不知道应该怎么修改，谢谢

train_data = read_data(train_record, train_batch_size, True, 10)

if train_steps % 500 == 0:

**Comment by liyibo** [comment]
@tectal 因为我不太熟悉tfrecord的使用，训练时候确实会出现上述的 “OutOfRangeError”，
“read_data(train_record, train_batch_size, True, 10)” 中数据复制的次数，可以看做是训练时候 epoch 的次数，你改为10次的话，应该是能看到模型在 val 数据集上的结果的，达到 epoch 的次数，上述错误就会出现，这个错误不影响模型保存和测试的，如果不想让这样的错误出现，可以将 tfrecord 数据读取放到 try 语句中，修改模型训练的那段代码

**Comment by dengxiaotian123** [comment]
Hello ,I want to know which tensorflow version is
